plugin/action/throttle/redis_limiter.go

```diff
@@ -0,0 +1,181 @@
+package throttle
+
+import (
+	"bytes"
+	"context"
+	"strconv"
+	"time"
+
+	"github.com/ozontech/file.d/logger"
+	"github.com/ozontech/file.d/pipeline"
+)
+
+const (
+	keySuffix = "limit"
+)
+
+type redisLimiter struct {
+	redis redisClient
+
+	// <keyPrefix>_
+	// will be used for bucket counter <keyPrefix>_<bucketID> and limit key <keyPrefix>_limit
+	keyPrefix bytes.Buffer
+
+	// contains values which will be used for incrementing remote bucket counter
+	// buckets will be flushed after every sync to contain only increment value
+	incrementLimiter *inMemoryLimiter
+
+	// contains global values synced from redis
+	totalLimiter *inMemoryLimiter
+}
+
+// NewRedisLimiter return instance of redis limiter.
+func NewRedisLimiter(
+	ctx context.Context,
+	redis redisClient,
+	pipelineName, throttleFieldName, throttleFieldValue string,
+	bucketInterval time.Duration,
+	bucketCount int,
+	limit complexLimit,
+) *redisLimiter {
+	rl := &redisLimiter{
+		redis:            redis,
+		incrementLimiter: NewInMemoryLimiter(bucketInterval, bucketCount, limit),
+		totalLimiter:     NewInMemoryLimiter(bucketInterval, bucketCount, limit),
+	}
+
+	rl.keyPrefix = bytes.Buffer{}
+
+	// full name of keyPrefix will be pipelineName_throttleFieldName_throttleFieldValue_limit. `limit` added afterwards
+	rl.keyPrefix.WriteString(pipelineName)
+	rl.keyPrefix.WriteString("_")
+	rl.keyPrefix.WriteString(throttleFieldName)
+	rl.keyPrefix.WriteString("_")
+	rl.keyPrefix.WriteString(throttleFieldValue)
+	rl.keyPrefix.WriteString("_")
+
+	return rl
+}
+
+func (l *redisLimiter) isAllowed(event *pipeline.Event, ts time.Time) bool {
+	// count increment from last sync
+	isAllowed := l.incrementLimiter.isAllowed(event, ts)
+	if isAllowed {
+		// count global limiter if distributed mode on and passed local limit
```

Comment:
where is no "distributed" mode in the settings, make comment actual

Reply:
Good point